spec/services/booking_request_creator_spec.rb

```diff
@@ -54,13 +54,13 @@
         slot_option_1: '2015-01-02T09:00/10:00',
         slot_option_2: '2015-01-03T09:00/10:00',
         slot_option_3: '2015-01-04T09:00/10:00'
-      )
+      ).and_return double(Visit, id: 2)
 
     described_class.new.create! prisoner_step, visitors_step, slots_step
   end
 
   it 'emails the prison' do
-    visit = double(Visit)
+    visit = double(Visit, id: 2)
     allow(Visit).to receive(:create!).and_return(visit)
```

Comment:
Could you DRY this and the visit declaration out, please? I'd suggest something like: 

```
context 'emailing and logging' do 
  let(:visit) { double(Visit, id: 2) }
  before do
    allow(Visit).to receive(:create!).and_return(visit)
  end

  it 'emails the prison' do 
    expect(PrisonMailer).to receive(:request_received).with(visit)...
   ...

 it 'logs the visit id' do 
   expect(LogStasher.request_context).to match(visit_id: 2) 
```